src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1397,6 +1397,33 @@ namespace cryptonote
     return res;
   }
   //-----------------------------------------------------------------------------------------------
+
+  bool core::check_storage_server_ping() const
+  {
+    CRITICAL_REGION_LOCAL(m_ping_lock);
```

Comment:
Similar here, are we trying to future proof this by pre-emptively locking it, incase this function gets called from more places than we expect? If not, then since calls are serialized to on_idle- no thread races.

Even so, the race would be to read a value that might be slightly out of date by milliseconds.

Or I suppose scarier is std::chrono's timepoint not thread safe?

Reply:
`update_storage_server_last_ping` is called on an RPC request, which is not serialised with on_idle calls. I assume `time_point` is "thread-safe", but I couldn't confirm that. There is always the problem of instruction reordering, but for this simple case that shouldn't matter. That said, I don't see any real downside to protecting with a mutex.

Reply:
I see, yeah misunderstood that this is called via RPC not on_idle, which the handler could be single threaded, but nevertheless, this would be a good place for an atomic since we're  synchronizing access to a single variable.

Reply:
Oh wait but that means you can't use time_point .. :/ I vote for time_t

Reply:
OK, changed to `atomic<time_t>`